internal/gps/source_manager.go

```diff
@@ -153,21 +154,38 @@ func NewSourceManager(cachedir string) (*SourceMgr, error) {
 		return nil, err
 	}
 
+	// Fix for #820
+	//
+	// Consult https://godoc.org/github.com/nightlyone/lockfile for the lockfile
+	// behaviour. It's magic. It deals with stale processes, and if there is
+	// a process keeping the lock busy, it will pass back a temporary error that
+	// we can spin on.
+
 	glpath := filepath.Join(cachedir, "sm.lock")
-	_, err = os.Stat(glpath)
-	if err == nil {
+	lockfile, err := lockfile.New(glpath)
+	if err != nil {
 		return nil, CouldNotCreateLockError{
```

Comment:
Slightly related to this PR: why do we define `CouldNotCreateLockError` as a struct, with `Path` when it's always included in `Err`?

I recommend we update `CouldNotCreateLockError.Error()` to allow:

```go
return nil, CouldNotCreateLockError{
    Path: glpath,
    Err:  err,
}
```

by doing the following:

```go
func (e CouldNotCreateLockError) Error() string {
	fmt.Errorf("unable to create lock %s: %s", glpath, err)
}
```